Gifski/Preview/PreviewRenderer.swift

```diff
@@ -0,0 +1,97 @@
+import CoreImage
+import AppKit
+import CoreGraphics
+
+struct PreviewRenderer {
+	struct PreviewCheckerboardParameters: Equatable {
+		let isDarkMode: Bool
+		let videoBounds: CGRect
+	}
+
+	static func renderPreview(
+		previewFrame: CVPixelBuffer,
+		outputFrame: CVPixelBuffer,
+		previewCheckerboardParams: PreviewCheckerboardParameters
+	) async throws {
+		let previewImage = CIImage(
+			cvPixelBuffer: previewFrame,
+			options: outputFrame.colorSpace.map { space -> [CIImageOption: Any] in
+				[
+					.colorSpace: space
+				]
+			}
+		)
+		try await renderPreview(previewImage: previewImage, outputFrame: outputFrame, previewCheckerboardParams: previewCheckerboardParams)
+	}
+
+	static func renderPreview(
+		previewFrame: CGImage,
+		outputFrame: CVPixelBuffer,
+		previewCheckerboardParams: PreviewCheckerboardParameters
+	) async throws {
+		try await renderPreview(previewImage: CIImage(cgImage: previewFrame), outputFrame: outputFrame, previewCheckerboardParams: previewCheckerboardParams)
+	}
+
+	private static func renderPreview(
+		previewImage: CIImage,
+		outputFrame: CVPixelBuffer,
+		previewCheckerboardParams: PreviewCheckerboardParameters
+	) async throws {
+		let context = CIContext()
+		let outputWidth = Double(outputFrame.width)
+		let outputHeight = Double(outputFrame.height)
+		let outputSize = CGSize(width: outputWidth, height: outputHeight)
+		let outputRect = CGRect(origin: .zero, size: outputSize)
+
+		let checkerboard = createCheckerboard(
+			outputRect: outputRect,
+			uniforms: previewCheckerboardParams
+		)
+
+		let previewBounds = previewImage.extent
+
+		let translationX = (outputWidth - previewBounds.width) / 2 - previewBounds.minX
+		let translationY = (outputHeight - previewBounds.height) / 2 - previewBounds.minY
+
+		let transform = CGAffineTransform.identity
+			.translatedBy(x: translationX, y: translationY)
+
+		let translatedPreview = previewImage.transformed(by: transform)
+		let result = translatedPreview.composited(over: checkerboard)
+
+		context.render(
+			result.cropped(to: outputRect),
+			to: outputFrame,
+			bounds: outputRect,
+			colorSpace: outputFrame.colorSpace ?? CGColorSpace(name: CGColorSpace.sRGB)
+		)
+	}
+
+	private static func createCheckerboard(
+		outputRect: CGRect,
+		uniforms: PreviewCheckerboardParameters
+	) -> CIImage {
+		guard let filter = CIFilter(name: "CICheckerboardGenerator") else {
```

Comment:
Use `CIFilter.checkerboardGenerator()`. You must import `import CoreImage.CIFilterBuiltins` instead then.